### ISSUE TITLE
nuget.exe "pack" should re-evaluate project versions to allow git versioning for packages.config

### ISSUE THREAD

**Issue by theschnarr** [NuGet/Home#6839]
I'm running into the same issue reported in [#4790](https://github.com/NuGet/Home/issues/4790).
Essentially, I have two projects in the same solution (A and B).  B references A.  Both have nuspec files, and both are using GitVersionTask to generate a SemVer.

When I use Nuget to pack project B, using includereferencedprojects, A shows as a dependency in the output package for B, but is using 1.0.0, instead of the generated version number.

Base on the history from #4790, this issue may have been fixed for the new csproj and msbuild /t: pack, but I am still running into this situation with the traditional packages.config, and traditional .NET Class Library project.

NuGet product used (NuGet.exe)

NuGet version (4.6.2.5055):

VS version (if appropriate): 15.6.6 (VS 2017 Pro)

OS version (i.e. win10 v1607 (14393.321)): win10 v1709

## Detailed repro steps so we can see the same problem

1. Create two .NET Framework (legacy csproj) libraries: A and B with a common solution directory.
2. Add the solution to git source control and commit.
3. Add reference from B to A
4. Install GitVersionTask package to both
5. Commit all changes to git.
6. Run a package restore on the solution and pack both projects.

I've tried the workaround from #4790, but it does not seem to have any affect when using nuget cli and packages.config.

Any help/suggestions would be greatly appreciated.

**Comment by rohit21agrawal** [comment]
yes this scenario currently does not work with packages.config + nuget.exe. I do not currently have an ETA for when this would be fixed, but if you can move over to NETCore style SDK projects (or move to PackageReference atleast), that would be the fastest way to workaround the issue.
